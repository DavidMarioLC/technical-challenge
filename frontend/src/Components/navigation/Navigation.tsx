import './navigation.css';
import {BsBook} from 'react-icons/bs'
import {RiGitRepositoryLine} from 'react-icons/ri'
import {AiOutlineProject} from 'react-icons/ai'
import {FiPackage} from "react-icons/fi"

interface Props {
  total_repositories:number
}

const Navigation = ({total_repositories}:Props) => {
  return (
    <nav className="navigation">
      <ul className="navigation__list">
        <li className="navigation__item">
          <BsBook className='navigation__icon'/> 
         <span>Overview</span>
        </li>
        <li className="navigation__item navigation__item--active">
          <RiGitRepositoryLine className='navigation__icon'/>
          <span>Repositories</span>
          <span className='navigation__repositories'>{total_repositories}</span>
        </li>
        <li className="navigation__item">
          <AiOutlineProject className='navigation__icon'/>
         <span>Projects</span>
        </li>
        <li className="navigation__item">
          <FiPackage className='navigation__icon'/>
         <span>Packages</span>
        </li>
      </ul>
    </nav>
  )
}

export default Navigation
