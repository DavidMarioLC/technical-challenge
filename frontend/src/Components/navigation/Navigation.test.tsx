import renderer from 'react-test-renderer';
import Navigation from './Navigation';


it('render Navigation correctly', () => {
  const tree = renderer.create(<Navigation total_repositories={0}/>).toJSON();
  expect(tree).toMatchSnapshot();
});