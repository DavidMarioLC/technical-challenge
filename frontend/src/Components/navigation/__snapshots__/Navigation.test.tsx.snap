// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Navigation correctly 1`] = `
<nav
  className="navigation"
>
  <ul
    className="navigation__list"
  >
    <li
      className="navigation__item"
    >
      <svg
        className="navigation__icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 2.828c.885-.37 2.154-.769 3.388-.893 1.33-.134 2.458.063 3.112.752v9.746c-.935-.53-2.12-.603-3.213-.493-1.18.12-2.37.461-3.287.811V2.828zm7.5-.141c.654-.689 1.782-.886 3.112-.752 1.234.124 2.503.523 3.388.893v9.923c-.918-.35-2.107-.692-3.287-.81-1.094-.111-2.278-.039-3.213.492V2.687zM8 1.783C7.015.936 5.587.81 4.287.94c-1.514.153-3.042.672-3.994 1.105A.5.5 0 0 0 0 2.5v11a.5.5 0 0 0 .707.455c.882-.4 2.303-.881 3.68-1.02 1.409-.142 2.59.087 3.223.877a.5.5 0 0 0 .78 0c.633-.79 1.814-1.019 3.222-.877 1.378.139 2.8.62 3.681 1.02A.5.5 0 0 0 16 13.5v-11a.5.5 0 0 0-.293-.455c-.952-.433-2.48-.952-3.994-1.105C10.413.809 8.985.936 8 1.783z"
        />
      </svg>
      <span>
        Overview
      </span>
    </li>
    <li
      className="navigation__item navigation__item--active"
    >
      <svg
        className="navigation__icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M13 21v2.5l-3-2-3 2V21h-.5A3.5 3.5 0 0 1 3 17.5V5a3 3 0 0 1 3-3h14a1 1 0 0 1 1 1v17a1 1 0 0 1-1 1h-7zm0-2h6v-3H6.5a1.5 1.5 0 0 0 0 3H7v-2h6v2zm6-5V4H6v10.035A3.53 3.53 0 0 1 6.5 14H19zM7 5h2v2H7V5zm0 3h2v2H7V8zm0 3h2v2H7v-2z"
            fillRule="nonzero"
          />
        </g>
      </svg>
      <span>
        Repositories
      </span>
      <span
        className="navigation__repositories"
      >
        0
      </span>
    </li>
    <li
      className="navigation__item"
    >
      <svg
        className="navigation__icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M280 752h80c4.4 0 8-3.6 8-8V280c0-4.4-3.6-8-8-8h-80c-4.4 0-8 3.6-8 8v464c0 4.4 3.6 8 8 8zm192-280h80c4.4 0 8-3.6 8-8V280c0-4.4-3.6-8-8-8h-80c-4.4 0-8 3.6-8 8v184c0 4.4 3.6 8 8 8zm192 72h80c4.4 0 8-3.6 8-8V280c0-4.4-3.6-8-8-8h-80c-4.4 0-8 3.6-8 8v256c0 4.4 3.6 8 8 8zm216-432H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
        />
      </svg>
      <span>
        Projects
      </span>
    </li>
    <li
      className="navigation__item"
    >
      <svg
        className="navigation__icon"
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="16.5"
          x2="7.5"
          y1="9.4"
          y2="4.21"
        />
        <path
          d="M21 16V8a2 2 0 0 0-1-1.73l-7-4a2 2 0 0 0-2 0l-7 4A2 2 0 0 0 3 8v8a2 2 0 0 0 1 1.73l7 4a2 2 0 0 0 2 0l7-4A2 2 0 0 0 21 16z"
        />
        <polyline
          points="3.27 6.96 12 12.01 20.73 6.96"
        />
        <line
          x1="12"
          x2="12"
          y1="22.08"
          y2="12"
        />
      </svg>
      <span>
        Packages
      </span>
    </li>
  </ul>
</nav>
`;
